### PR TITLE
Refactor the serde logic of JTensor and Sample

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?

Previously we pass JTensor from python to java via ArrayList which would result in bad performance.
After this changes we would use Array[Byte] directly for those transforming.

## How was this patch tested?
unittest

### CODE REVIEW

**Comment by zhichao-li** [comment]
Jenkins passed. http://172.168.2.101:8080/view/PR-Validation/job/BigDL-PR-Validation/1712/

**Comment by zhichao-li** [comment]
cc @jenniew

**Comment by zhichao-li** [comment]
cc @jason-dai  @yiheng  any comments? I would merge it if no objections.

File: pyspark/bigdl/util/common.py [hunk 1]
```diff
@@ -148,75 +154,69 @@ def from_ndarray(cls, a_ndarray, bigdl_type="float"):
...
+        :param bigdl_type: "double" or "float"
+        """
+        self.features = features
+        self.label = label
         self.bigdl_type = bigdl_type
 
     @classmethod
     def from_ndarray(cls, features, label, bigdl_type="float"):
+        """
+        Convert a ndarray to Tensor which would be used in Java side.
```
Comment (jason-dai) [inline pyspark/bigdl/util/common.py]: to Sample?

**Pull request merged**